### PR TITLE
Add ledger total count api

### PR DESCRIPTION
**Pull request by DanielZhangReal**
## Description

Add ledger total count api

Goes together with https://github.com/Manta-Network/sdk/pull/128

Refer to this PR: https://github.com/Manta-Network/Manta/pull/1109

---

Before we can approve this PR for merge, please make sure that **all** the following items have been checked off:
- [ ] Connected to an issue with discussion and accepted design using zenhub "Connect issue" button below
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [x] Explicitly labelled `A-calamari` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [x] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.

### CODE REVIEW

File: pallets/manta-sbt/src/lib.rs [hunk 1]
```diff
@@ -885,21 +897,11 @@ where
...
-        let mut receivers_total = (0..=255)
-            .map(|i| ShardTrees::<T>::get(i).current_path.leaf_index as u128)
-            .sum::<u128>();
-        // Workaround for the fact that we index the receivers from 0
-        if receivers_total == 0u128 {
-            // The cast is fine because the vector sizes are limited by PULL_MAX_RECEIVER_UPDATE_SIZE
-            receivers_total = receivers.len() as u128;
-        } else {
-            receivers_total += 256u128;
-        }
```
Comment (ghzlatarev) [inline pallets/manta-sbt/src/lib.rs:897]: where is this part ?

Comment (DanielZhangReal) [reply pallets/manta-sbt/src/lib.rs:897]: // In the initial state of the ledger, the total value will be 256;
// if we want to get an accurate value, we need to request `pull_receivers` to fix this value;
// but a simple total count interface does not need to add more complicated logic.

I added some notes on `pull_ledger_total_count`

Comment (ghzlatarev) [reply pallets/manta-sbt/src/lib.rs:897]: i'd leave the logic here, to make it easier on the users.

Comment (DanielZhangReal) [reply pallets/manta-sbt/src/lib.rs:897]: Make sure `pull_ledger_total_count` is consistent with `senders_receivers_total` returned by `pull_ledger_diff`, directly returns the value of `pull_ledger_total_count`

Comment (ghzlatarev) [reply pallets/manta-sbt/src/lib.rs:897]: Sorry i meant where is the other part `(if total_senders_receivers == 256) total_senders_receivers = receivers.count()`

Comment (DanielZhangReal) [reply pallets/manta-sbt/src/lib.rs:897]: I removed this logic, The above explains why we don't need this logic

Comment (DanielZhangReal) [reply pallets/manta-sbt/src/lib.rs:897]: I don’t think it’s necessary to add that logic for a new chain with only a few ledger data. If you think we need to add, then I may need to request `pull_receivers` on the `pull_ledger_total_count` interface to be compatible with the correct data, as I said above.

Comment (ghzlatarev) [reply pallets/manta-sbt/src/lib.rs:897]: So you're essentially leaving to the caller ?
It seems better to do the correction here, to make it easier for the users of this RPC. That was the idea initially.
Or am i missing something else ?

Comment (DanielZhangReal) [reply pallets/manta-sbt/src/lib.rs:897]: This is what I considered when adding `pull_ledger_total_count`

Comment (ghzlatarev) [reply pallets/manta-sbt/src/lib.rs:897]: im fine either way, let's see what the others say

Comment (Apokalip) [reply pallets/manta-sbt/src/lib.rs:897]: I think either way is fine, just make sure to optimize easier usage of RPCs

Comment (ferrell-code) [reply pallets/manta-sbt/src/lib.rs:897]: Looks fine to me, early value doesn't seem to be relevant.

**Review by ferrell-code — COMMENTED** [review]
Looks good to me.

Need to make ci happy

**Pull request closed without merging**